SimTracker/SiStripDigitizer/plugins/SiStripApvSimulationParametersESProducer.cc

```diff
@@ -0,0 +1,121 @@
+#include "FWCore/Framework/interface/ESProducer.h"
+#include "FWCore/Framework/interface/ModuleFactory.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "CondFormats/DataRecord/interface/SiStripCondDataRecords.h"
+#include "CondFormats/SiStripObjects/interface/SiStripApvSimulationParameters.h"
+#include <fstream>
+#include <boost/range/adaptor/indexed.hpp>
+
+class SiStripApvSimulationParametersESProducer : public edm::ESProducer {
+public:
+  explicit SiStripApvSimulationParametersESProducer(const edm::ParameterSet& conf);
+  ~SiStripApvSimulationParametersESProducer() override {}
+
+  std::unique_ptr<SiStripApvSimulationParameters> produce(const SiStripApvSimulationParametersRcd& record);
+
+private:
+  std::vector<edm::FileInPath> baselineFiles_TOB_;
+  std::vector<edm::FileInPath> baselineFiles_TIB_;
+  unsigned int baseline_nBins_;
+  float baseline_min_;
+  float baseline_max_;
+  std::vector<float> puBinEdges_;
+  std::vector<float> zBinEdges_;
+
+  SiStripApvSimulationParameters::LayerParameters makeLayerParameters(const std::string& apvBaselinesFileName) const;
+};
+
+SiStripApvSimulationParametersESProducer::SiStripApvSimulationParametersESProducer(const edm::ParameterSet& conf)
+    : baseline_nBins_(conf.getParameter<unsigned int>("apvBaselines_nBinsPerBaseline")),
+      baseline_min_(conf.getParameter<double>("apvBaselines_minBaseline")),
+      baseline_max_(conf.getParameter<double>("apvBaselines_maxBaseline")) {
+  setWhatProduced(this);
+  for (const auto x : conf.getParameter<std::vector<double>>("apvBaselines_puBinEdges")) {
+    puBinEdges_.push_back(x);
+  }
+  for (const auto x : conf.getParameter<std::vector<double>>("apvBaselines_zBinEdges")) {
+    zBinEdges_.push_back(x);
+  }
+  baselineFiles_TIB_ = {conf.getParameter<edm::FileInPath>("apvBaselinesFile_tib1"),
```

Comment:
@pieterdavid @EmyrClement , but then let me get this straight. The idea is still to read the baseline files from ASCII file and then populate `SiStripApvSimulationParametersRcd` via ESProducer?
If we store the object in DB one could retrieve it directly from Global Tag without resorting to have ASCII files and  ESProducers  around.

Reply:
I think we should store it in the DB indeed (so we don't need the ESProducer in production), but with an ESProducer it's still possible to run directly with the ASCII files when developing/validating them, and it can be used to put them into the DB with a very simple EDAnalyzer.

Reply:
Agreed.
One caveat: once the payload is read from DB we should be careful not to create dependencies on this object on all the GTs for all workflows that run the strip digitizer. One should make sure the APV dyamic gain is called only within the `Era_Run2_2016` so that only one GT develops the dependency.

Reply:
Indeed, that's why I changed the SiStripDigitizer to read the 'includeAPVSimulation' flag and only retrieve the product from the EventSetup if that is true.